File: mixql-engine/src/main/scala/org/mixql/engine/core/IModuleExecutor.scala
package org.mixql.engine.core

import org.mixql.protobuf.messages

trait IModuleExecutor {

  def reactOnExecute(msg: messages.Execute)(implicit
                                            identity: String,
                                            clientAddress: String
  ): messages.Message

  def reactOnSetParam(msg: messages.SetParam)(implicit
                                              identity: String,
                                              clientAddress: String
  ): messages.ParamWasSet

  def reactOnGetParam(msg: messages.GetParam)(implicit
                                              identity: String,
                                              clientAddress: String
  ): messages.Message

  def reactOnIsParam(msg: messages.IsParam)(implicit
                                            identity: String,
                                            clientAddress: String
  ): messages.Bool

  def reactOnShutDown()(implicit
                        identity: String,
                        clientAddress: String
  ): Unit = {}

  def reactOnExecuteFunction(msg: messages.ExecuteFunction)(implicit
                                                            identity: String,
                                                            clientAddress: String
  ): messages.Message

  def reactOnGetDefinedFunctions()(implicit
                                   identity: String,
                                   clientAddress: String
  ): messages.DefinedFunctions
}

File: mixql-engine/src/main/java/org/mixql/engine/core/IModuleExecutor.java
package org.mixql.engine.core;

import org.mixql.engine.core.logger.ModuleLogger;
import org.mixql.remote.messages.*;
import org.mixql.remote.messages.module.DefinedFunctions;
import org.mixql.remote.messages.client.Execute;
import org.mixql.remote.messages.client.ExecuteFunction;

public interface IModuleExecutor {

    Message reactOnExecuteAsync(Execute msg, String identity, String clientAddress, ModuleLogger logger,
                                PlatformContext platformContext);

    void reactOnShutDown(String identity, String clientAddress, ModuleLogger logger);

    Message reactOnExecuteFunctionAsync(ExecuteFunction msg, String identity, String clientAddress, ModuleLogger logger,
                                        PlatformContext platformContext);

    DefinedFunctions reactOnGetDefinedFunctions(String identity, String clientAddress, ModuleLogger logger);
}
